Foundation/HTTPCookieStorage.swift

```diff
@@ -36,14 +36,34 @@ extension HTTPCookie {
 */
 open class HTTPCookieStorage: NSObject {
 
+    /* both sharedStorage and sharedCookieStorages are synchronized on sharedSyncQ */
     private static var sharedStorage: HTTPCookieStorage?
     private static var sharedCookieStorages: [String: HTTPCookieStorage] = [:] //for group storage containers
+    private static let sharedSyncQ = DispatchQueue(label: "org.swift.HTTPCookieStorage.sharedSyncQ")
+
+    /* only modified in init */
     private var cookieFilePath: String!
-    private let workQueue: DispatchQueue = DispatchQueue(label: "HTTPCookieStorage.workqueue")
-    var allCookies: [String: HTTPCookie]
+
+    /* synchronized on syncQ, please don't use _allCookies directly outside of init/deinit */
+    private var _allCookies: [String: HTTPCookie]
+    private var allCookies: [String: HTTPCookie] {
+        get {
+            if #available(OSX 10.12, iOS 10.0, tvOS 10.0, watchOS 3.0, *) {
```

Comment:
what about linux?

Reply:
@phausler Linux seems to be handled in the `*`:

```
$ cat test.swift
if #available(OSX 10.12, iOS 10.0, tvOS 10.0, watchOS 3.0, *) {
    print("yes")
} else {
    print("no")
}
$ swift test.swift
yes
```

Reply:
I agree to @weissi